CONTRIBUTING.md

```diff
@@ -0,0 +1 @@
+# Developer Guide
```

Comment:
We can have a  dedicated PR for the contribution guide; it's fine to have it just a placeholder for the time being.